### IMDB Project, Part 2
#### Laura Dimaria
8/1/22

Note: I am only able to extract year 2001, so I completed the assignments with the year I had and will come back when the issue is fixed.

In [2]:
import pandas as pd
import numpy as np

In [4]:
year_2000 = pd.read_csv('/Users/lauradimaria/Desktop/GitHub/CD--Project-3-IMDB-/Data/final_tmdb_data_2000.csv')
year_2000 = year_2000.drop(labels=0, axis=0)
year_2000.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   
5  tt0118852    0.0                               NaN                   NaN   

       budget                                             genres  \
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
5         0.0                      [{'id': 18, 'name': 'Drama'}]   

                               homepage        id original_language  \
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   
5                                   NaN   49511.0                en   

    original_title  ...     revenue  runtime  \
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   
5   Chinese Coffee  ...         0.0     99.0   

                                    spoken_languages    status  \
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
5  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                 title  \
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   
5  There's a fine line between friendship and bet...        Chinese Coffee   

   video  vote_average vote_count certification  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         0.000        0.0           NaN  
4    0.0         8.106     1925.0            PG  
5    0.0         6.900       46.0             R  

[5 rows x 26 columns]

In [5]:
year_2001 = pd.read_csv('/Users/lauradimaria/Desktop/GitHub/CD--Project-3-IMDB-/Data/final_tmdb_data_2001.csv.gz')
year_2001 = year_2001.drop(labels=0, axis=0)
year_2001.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1285 entries, 1 to 1285
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                1285 non-null   object 
 1   adult                  1285 non-null   float64
 2   backdrop_path          649 non-null    object 
 3   belongs_to_collection  86 non-null     object 
 4   budget                 1285 non-null   float64
 5   genres                 1285 non-null   object 
 6   homepage               108 non-null    object 
 7   id                     1285 non-null   float64
 8   original_language      1285 non-null   object 
 9   original_title         1285 non-null   object 
 10  overview               1257 non-null   object 
 11  popularity             1285 non-null   float64
 12  poster_path            1128 non-null   object 
 13  production_companies   1285 non-null   object 
 14  production_countries   1285 non-null   object 
 15  rele

In [6]:
#combine 2000 and 2001 dataframes into one
years_combined = pd.concat([year_2000, year_2001])
years_combined.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   
5  tt0118852    0.0                               NaN                   NaN   

       budget                                             genres  \
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...   
5         0.0                      [{'id': 18, 'name': 'Drama'}]   

                               homepage        id original_language  \
1                                   NaN   62127.0                en   
2                                   NaN  110977.0                en   
3                                   NaN  442869.0                hi   
4  http://www.wkw-inthemoodforlove.com/     843.0                cn   
5                                   NaN   49511.0                en   

    original_title  ...     revenue  runtime  \
1  The Fantasticks  ...         0.0     86.0   
2    For the Cause  ...         0.0    100.0   
3             Gang  ...         0.0    152.0   
4             花樣年華  ...  12854953.0     99.0   
5   Chinese Coffee  ...         0.0     99.0   

                                    spoken_languages    status  \
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   
5  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   

                                             tagline                 title  \
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   
5  There's a fine line between friendship and bet...        Chinese Coffee   

   video  vote_average vote_count certification  
1    0.0         5.500       22.0           NaN  
2    0.0         5.100        8.0           NaN  
3    0.0         0.000        0.0           NaN  
4    0.0         8.106     1925.0            PG  
5    0.0         6.900       46.0             R  

[5 rows x 26 columns]

In [7]:
years_combined.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2488 entries, 1 to 1285
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                2488 non-null   object 
 1   adult                  2488 non-null   float64
 2   backdrop_path          1276 non-null   object 
 3   belongs_to_collection  196 non-null    object 
 4   budget                 2488 non-null   float64
 5   genres                 2488 non-null   object 
 6   homepage               174 non-null    object 
 7   id                     2488 non-null   float64
 8   original_language      2488 non-null   object 
 9   original_title         2488 non-null   object 
 10  overview               2436 non-null   object 
 11  popularity             2488 non-null   float64
 12  poster_path            2187 non-null   object 
 13  production_companies   2488 non-null   object 
 14  production_countries   2488 non-null   object 
 15  rele

### How many movies had at least some valid financial information (values > 0 for budget OR revenue)?

In [8]:
years_combined = years_combined[(years_combined['revenue'] > 0.0) | (years_combined['budget'] > 0.0)]
years_combined.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 617 entries, 1 to 1265
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                617 non-null    object 
 1   adult                  617 non-null    float64
 2   backdrop_path          514 non-null    object 
 3   belongs_to_collection  102 non-null    object 
 4   budget                 617 non-null    float64
 5   genres                 617 non-null    object 
 6   homepage               80 non-null     object 
 7   id                     617 non-null    float64
 8   original_language      617 non-null    object 
 9   original_title         617 non-null    object 
 10  overview               614 non-null    object 
 11  popularity             617 non-null    float64
 12  poster_path            596 non-null    object 
 13  production_companies   617 non-null    object 
 14  production_countries   617 non-null    object 
 15  relea

617 movies had valid financial information in either budget or revenue, down from 1284.

### Drop remainder movies with 0's for budget AND revenue

In [9]:
years_combined.drop(years_combined.loc[years_combined['revenue'] == 0.0].index, inplace=True)
years_combined.drop(years_combined.loc[years_combined['budget'] == 0.0].index, inplace=True)
years_combined.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 306 entries, 11 to 1242
Data columns (total 26 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   imdb_id                306 non-null    object 
 1   adult                  306 non-null    float64
 2   backdrop_path          295 non-null    object 
 3   belongs_to_collection  76 non-null     object 
 4   budget                 306 non-null    float64
 5   genres                 306 non-null    object 
 6   homepage               47 non-null     object 
 7   id                     306 non-null    float64
 8   original_language      306 non-null    object 
 9   original_title         306 non-null    object 
 10  overview               306 non-null    object 
 11  popularity             306 non-null    float64
 12  poster_path            306 non-null    object 
 13  production_companies   306 non-null    object 
 14  production_countries   306 non-null    object 
 15  rele

306 movies have some financial data for both revenue and budget

### How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [10]:
years_combined['certification'].value_counts()

R        122
PG-13    102
PG        23
G         11
NR         6
Name: certification, dtype: int64

### What is the average revenue per certification category?

In [11]:
#code stolen from stack overflow
#link: https://stackoverflow.com/questions/21137150/format-suppress-scientific-notation-from-pandas-aggregation-results
pd.set_option('display.float_format', lambda x: f'{x:,.3f}')

years_combined.groupby('certification')['revenue'].mean()

certification
G       146,353,970.636
NR       23,823,316.000
PG      143,951,020.348
PG-13   105,875,121.735
R        51,657,227.525
Name: revenue, dtype: float64

### What is the average budget per certification category?

In [12]:
pd.set_option('display.float_format', lambda x: f'{x:,.3f}')
years_combined.groupby('certification')['budget'].mean()

certification
G       47,000,000.000
NR      14,783,333.333
PG      54,617,391.696
PG-13   45,984,288.108
R       26,588,639.754
Name: budget, dtype: float64

In [13]:
#Save DF to csv

years_combined.to_csv('Data/tmdb_results_combined.csv.gz',compression= "gzip", index= False)